### ISSUE TITLE
The UpdatedODataResult default response status code should not be 204

### ISSUE THREAD

**Issue by LianwMS** [OData/WebApi#125]
The protocol says:  

```
9.1.4 Response Code 204 No Content
A request returns 204 No Content if the requested resource has the null value, or if the service applies a return=minimal preference. In this case, the response body MUST be empty.
```

But if an action returns Updated(...) in a controller, then the default
response code is 204. Actually it should be 200 unless the Prefer header of
"return=minimal" is in the request.  
## Work Item Details

**Original CodePlex Issue:**    [Issue 2036](http://aspnetwebstack.codeplex.com/workitem/2036)
**Status:** Proposed
**Reason Closed:**  Unassigned
**Assigned to:**    Unassigned
**Reported on:**    Jun 27, 2014 at 9:31 AM
**Reported by:**    jinfutan
**Updated on:** Jul 31, 2014 at 10:16 PM
**Updated by:** rcollette

**Comment by LianwMS** [comment]
On *2014-06-27 16:39:02 UTC*, **jinfutan** commented:

And if the Prefer is applied, the response should contain Preference-Applied Header.

**Comment by LianwMS** [comment]
On *2014-07-01 02:03:03 UTC*, **yishaigalatzer** commented:

@jinfu - Please specify update + what value to be returned.

**Comment by LianwMS** [comment]
On *2014-07-01 17:02:05 UTC*, **jinfutan** commented:

If the header Prefer:return=minimal is in the request, then we should return 204 and the header Preference-Applied.

Or, we should return 200.

**Comment by LianwMS** [comment]
On *2014-07-03 11:26:33 UTC*, **azhou** commented:

It's a breaking change. We can't get it fixed in 5.3. 
Customer can specify return=minimal preference in the accept header. For return code of 200, we should return the non-empty payload.

**Comment by skottmckay** [comment]
Is this fixed by inverting this below test so that it is only if the caller specifies return-minimal that a 204 is returned?

Current [implementation ](https://github.com/OData/WebApi/blob/master/OData/src/System.Web.OData/OData/Results/UpdatedODataResult.cs) of GetInnerActionResult has

```
        internal IHttpActionResult GetInnerActionResult()
        {
            if (RequestPreferenceHelpers.RequestPrefersReturnContent(_innerResult.Request))
            {
                return _innerResult;
            }
            else
            {
                return new StatusCodeResult(HttpStatusCode.NoContent, _innerResult.Request);
            }
        }

```

vs. only returning 204 if explicitly requested 

```

        internal IHttpActionResult GetInnerActionResult()
        {
            if (RequestPreferenceHelpers.RequestPrefersReturnNoContent(_innerResult.Request))
            {
                return new StatusCodeResult(HttpStatusCode.NoContent, _innerResult.Request);
            }
            else
            {
                return _innerResult;
            }
        }
```
